packages/xod-client/src/core/selectors.js

```diff
@@ -1,8 +1,44 @@
+import R from 'ramda';
+import { Maybe } from 'ramda-fantasy';
+import { createSelector } from 'reselect';
+
+import * as XP from 'xod-project';
+
 import * as Editor from '../editor/selectors';
+import * as Project from '../project/selectors';
+import * as ProjectBrowser from '../projectBrowser/selectors';
 import * as Errors from '../messages/selectors';
 import * as Processes from '../processes/selectors';
 import * as Utils from '../utils/selectors';
 
+import { SELECTION_ENTITY_TYPE } from '../editor/constants';
+
+//
+// Docs sidebar
+//
+
+// TODO: naming
+export const getPatchForDocs = createSelector(
+  [
+    Project.getProject,
+    Editor.getSelection,
+    ProjectBrowser.getSelectedPatchPath,
+    Project.getCurrentPatchNodes,
+  ],
+  (project, editorSelection, selectedPatchPath, currentPatchNodes) => R.compose(
+    R.map(patchPath => XP.getPatchByPathUnsafe(patchPath, project)),
+    R.when(
+      Maybe.isNothing,
+      R.always(Maybe(selectedPatchPath))
+    ),
+    R.map(XP.getNodeType),
+    R.map(({ id }) => currentPatchNodes[id]),
+    Maybe,
+    R.find(R.propEq('entity', SELECTION_ENTITY_TYPE.NODE)),
+  )(editorSelection)
+);
```

Comment:
The logic is imperfect in some cases. I’ve actually thought I found a bug.

### Steps to reproduce

1. Open `welcome-to-xod` project
2. Select the `flip-flop` node
3. Press `h`
4. Expand `xod/core` in Project Browser
5. Click `acos`

### Expected behavior

Helpbar shows `xod/core/acos` info

### Actual behavior

Helpbar shows flip-flop info

### How to fix

Do we have a notion of focus? I.e. can we say whether Patch Board, Project Browser, or Inspector is in focus. If so, we should rely on it and show help for a node active in the focused pane.

Otherwise, maybe a quick workaround a-la “last click wins” would suffice I think.

Reply:
I implemented a basic notion of focus(sidebar or workarea), and now it's used to set priority of documentation source